eth/tracers/logger/logger.go

```diff
@@ -268,7 +268,9 @@ func (l *StructLogger) OnTxEnd(receipt *types.Receipt, err error) {
 		}
 		return
 	}
-	l.usedGas = receipt.GasUsed
+	if receipt != nil {
```

Comment:
I'm surprised you've seen a panic here. As far as I can see in state processor either receipt is nil and error is non-nil (in which case struct logger returns) or there is a receipt.

Reply:
Ah you're right. The issue issue is actually that `err` is redeclared.